tests/tel2puml/otel_to_pv/data_holders/sql_data_holder/test_sql_dataholder.py

```diff
@@ -703,3 +695,158 @@ def test_find_unique_graphs(
     assert unique_job_ids_per_job_name["test_name_2"] == {
         str(i) for i in range(5)
     }
+
+
+def test_stream_data(
+    sql_data_holder_with_multiple_otel_job_names: SQLDataHolder,
+) -> None:
+    """Test the stream_data function"""
+    sql_data_holder = sql_data_holder_with_multiple_otel_job_names
+
+    # Test 1: Stream all data
+    # Prep expected events
+    expected_events = {}
+    with sql_data_holder.session as session:
+        nodes = session.query(NodeModel).all()
+        for node in nodes:
+            otel_event = sql_data_holder.node_to_otel_event(node)
+            expected_events[otel_event.event_id] = otel_event
+
+    # Stream data and collect events
+    result = sql_data_holder.stream_data()
+    streamed_events = {}
+    for job_name, otel_event_gen in result:
+        for otel_event in otel_event_gen:
+            streamed_events[otel_event.event_id] = otel_event
+
+    # Compare the expected and streamed data
+    assert len(expected_events) == len(streamed_events)
+
+    for event_id, expected_event in expected_events.items():
+        assert event_id in streamed_events
+        assert expected_event == streamed_events[event_id]
+
+    # Test 2: Filter by job_name
+    result = sql_data_holder.stream_data(filter_job_names={"test_name_0"})
+    expected_job_names = ["test_name_0"]
+    all_events = []
+    for job_name, otel_event_gen in result:
+        assert job_name in expected_job_names
+        expected_job_names.remove(job_name)
+
+        for otel_event in otel_event_gen:
+            all_events.append(otel_event)
```

Comment:
think a check to make sure the otel event has the required job name would be good here